Formula/unit.rb

```diff
@@ -0,0 +1,95 @@
+class Unit < Formula
+  desc "Dynamic web and application server for multiple languages"
+  homepage "https://unit.nginx.org/"
+  url "https://unit.nginx.org/download/unit-1.1.tar.gz"
+  sha256 "7c66365f5ea87e7e8903bac9d639676947fe2ab7aa799ce35defc426b3409ee0"
+  head "https://hg.nginx.org/unit", :using => :hg
+
+  option "with-debug", "Build with debug logging enabled"
+
+  depends_on "openssl"
+  depends_on "pcre"
+
+  depends_on "go" => :optional
+  depends_on "perl" => :optional
+  depends_on "python" => :optional
+  depends_on "ruby" => :optional
+
+  def install
+    openssl = Formula["openssl"]
+    pcre = Formula["pcre"]
+
+    configure_args = [
+      "--prefix=#{prefix}",
+      "--bindir=#{bin}",
```

Comment:
Once prefix is chosen, let `bin` and `sbin` be determined as upstream prefers.

Reply:
This is again tricky. Unit is trying to put itself all into one directory, like an `/opt/unit`, which it also runs out of. I split it up a bit so it would link nicely into the `/usr/local` hierarchy, symmetric to the nginx formula and other unix-y daemon formulae, and make it easy to add shared objects in other formulae.

It's also a little tricky because most of these options are set at compile time, or on the command line — there is no config file.

A bare `./configure` (and perl, python, ruby) gives:

```
Configuration summary:

  unit bin directory:           "bin"
  unit sbin directory:          "sbin"
  unit modules directory:       "build"
  unit state directory:         "build"

  unit pid file:                "unit.pid"
  unit log file:                "unit.log"

  unit control API socket:      "unix:control.unit.sock"
```

and builds and installs into the current directory:

```
.
├── CHANGES
├── LICENSE
├── Makefile
├── NOTICE
├── README
├── auto
│   ├── atomic
│   ├── cc
│   └── ...
├── build
│   ├── Makefile
│   ├── autoconf.data
│   ├── autoconf.err
│   ├── ...
│   ├── perl.unit.so
│   ├── python.unit.so
│   ├── ruby.unit.so
│   ├── ...
│   ├── src
│   │   ├── nxt_app_log.dep
│   │   ├── nxt_app_log.o
│   │   ├── nxt_application.dep
│   │   ├── nxt_application.o
│   │   ├── ...
│   │   └── test
│   └── unitd
├── configure
├── sbin
│   └── unitd
├── src
│   ├── perl
│   │   └── ...
│   ├── nxt_aix_send_file.c
│   ├── nxt_app_log.c
│   ├── nxt_application.c
│   ├── nxt_application.h
│   ├── ...
│   └── test
│       ├── perl
│       │   └── ...
│       ├── nxt_gmtime_test.c
│       ├── nxt_http_parse_test.c
│       └── ...
└── test
    ├── perl
    │   └── ...
    ├── run.py
    ├── test_access_log.py
    └── ...

83 directories, 554 files
```

Just doing `./configure --prefix=...` (and perl, python, ruby) gives output like:

```
Configuration summary:

  unit bin directory:           "/usr/local/Cellar/nginx_unit/1.1/bin"
  unit sbin directory:          "/usr/local/Cellar/nginx_unit/1.1/sbin"
  unit modules directory:       "/usr/local/Cellar/nginx_unit/1.1/build"
  unit state directory:         "/usr/local/Cellar/nginx_unit/1.1/build"

  unit pid file:                "/usr/local/Cellar/nginx_unit/1.1/unit.pid"
  unit log file:                "/usr/local/Cellar/nginx_unit/1.1/unit.log"

  unit control API socket:      "unix:/usr/local/Cellar/nginx_unit/1.1/control.unit.sock"
```

and installs as a tree like:

```
/usr/local/Cellar/nginx_unit/1.1
├── build
│   ├── perl.unit.so
│   ├── python.unit.so
│   └── ruby.unit.so
├── sbin
│   └── unitd
├── unit.log
└── unit.pid
```

with all of its runtime stuff created within. There's nowhere to install shared modules from other formulae, logs are in unexpected places, etc.

Installing per the proposed formula give configure output:

```
Configuration summary:

  unit bin directory:           "/usr/local/Cellar/nginx_unit/1.1/bin"
  unit sbin directory:          "/usr/local/Cellar/nginx_unit/1.1/bin"
  unit modules directory:       "/usr/local/Cellar/nginx_unit/1.1/lib/unit"
  unit state directory:         "/usr/local/var/unit"

  unit pid file:                "/usr/local/var/run/unit.pid"
  unit log file:                "/usr/local/var/log/unit.log"

  unit control API socket:      "unix:/usr/local/var/run/unit.control.sock"

  non-privileged user:          "nobody"
  non-privileged group:         ""

  IPv6 support:                 YES
  Unix domain sockets support:  YES
  debug logging:                NO
```

and installs as a tree like:

```
├── bin
│   └── unitd
└── lib
    └── unit
        ├── perl.unit.so
        ├── python.unit.so
        └── ruby.unit.so
```

I'll see if I can get it to install the modules into `/usr/local/Cellar/nginx_unit/1.1/lib/unit` but load them from `/usr/local/lib/unit` for the multi-formulae approach.

Reply:
The proposed layout here is pretty close to their debian/rpm/docker packing, too:

https://github.com/nginx/unit/tree/master/pkg